src/writer/creator.cpp

```diff
@@ -501,7 +502,7 @@ namespace zim
       // If cluster will be too large, write it to dis, and open a new
       // one for the content.
       if ( cluster->count()
-        && cluster->size().v+itemSize >= minChunkSize * 1024
+        && cluster->size().v+itemSize >= clusterSize
```

Comment:
I think it makes sense to change the logic here. Consider the following scenario:

1. target cluster size == 1MiB ==1024KiB
2. the current incomplete cluster has very little data (say 10KiB)
3. adding the next (large) item will make it 1025KiB big

The current logic will close the cluster at its current small size, whereas slightly overshooting the target size wouldn't lead to anything particularly bad about that cluster. In fact it's better to have large items at the end of the cluster than at its start. In this case accessing the large item at the end will involve decompressing little additional data (and the overhead is small). Whereas if we push that large item to a new cluster, and then a small item is appended to it, accessing the latter will involve larger overhead.

Reply:
The logic here is to avoid to add huge file to uncompressed clusters.
This is especially the case with videos or xapian indexes.

Maybe we should use a different size for compressed and uncompressed cluster.
A "small" size for compressed clusters (which can be a bit overshoot)
A "big" size for uncompressed clusters (which can't be overshoot, except if a unique blob is bigger)

Reply:
I believe the size of uncompressed clusters does not matter that much. To me it could be constant and not tuneable.

Reply:
It matters, far less that for compressed cluster but yet. If we take the two extrems:
- 1 cluster per blob, we have the most size overhead (17 bytes per blob/cluster) and we need to parse the cluster all the time (we cache the cluster, but we have less chance that the cluster is already in the cache).
- 1 cluster with all blobs (17 bytes + 8 bytes per blob), but then we cache only one cluster but we will have to create N subreader for the blobs.

With a cluster size of 1Mb, we probably have very few blobs per uncompressed cluster. We can probably extend this. But I'm not sure about the value to use.

To resume, I agree with both of you.
- I will adapt the logic to close a cluster **after** it ovreshoot the target size.
- We can decide a have a different size for uncompressed cluster, but it would be better to do it in another PR.